hal/halx86/generic/dma.c

```diff
@@ -152,8 +152,8 @@ HalpInitDma(VOID)
         * Check if Extended DMA is available. We're just going to do a random
         * read and write.
         */
-        WRITE_PORT_UCHAR((PUCHAR)FIELD_OFFSET(EISA_CONTROL, DmaController2Pages.Channel2), 0x2A);
-        if (READ_PORT_UCHAR((PUCHAR)FIELD_OFFSET(EISA_CONTROL, DmaController2Pages.Channel2)) == 0x2A)
+        WRITE_PORT_UCHAR((PUCHAR)(ULONG_PTR)FIELD_OFFSET(EISA_CONTROL, DmaController2Pages.Channel2), 0x2A);
```

Comment:
I'm a little confused as to why these seem to have no base address, just the FIELD_OFFSET... but you didn't change that. The casts seem fine.